### PR TITLE
Result after remediaton

### LINKED ISSUE #39: Provide description of scanning results when it makes sense
If the scanner combines scanning and remediation, it may make sense to display context to scanning results.

This applies to cases when a rule scan fails, and remediation is executed.

- If the remediation fails (returns non-zero exit code, possibly produces an error message), it would make sense to accompany the FAIL result with "fix failed" or something like that.
- If the remediation succeeds, but the subsequent check fails, we display ERROR in the legacy oscap-produced reports. That's not really correct in terms of XCCDF, a FAIL with "fix unsuccessful" or something like that would describe the case better.
- If the remediation succeeds and the check also passes, we use FIXED - that's correct.

### PR DESCRIPTION
**Pull request by Honny1**
This PR contains:

* Fix missing message in rules
* Change result according to issue #39 

Fixes #39

### CODE REVIEW

File: oscap_report/html_report/templates/rule_detail.html [hunk 1]
```diff
@@ -142,19 +142,25 @@
...
                 <td role="cell">
+                    {% for message in rule.messages %}
                     <div class="pf-c-alert pf-m-info pf-m-inline">
                         <div class="pf-c-alert__icon">
                             <i class="fas fa-fw fa-info-circle" aria-hidden="true"></i>
                         </div>
                         <p class="pf-c-alert__title">
                             <span class="pf-screen-reader">Message:</span>
-                            {{ rule.message }}
+                            Message:
```
Comment (evgenyz) [inline oscap_report/html_report/templates/rule_detail.html:156]: I believe the result would be `Message: Message: The text of the message` in this case. Or I'm missing something here?

File: oscap_report/html_report/templates/js/interactive_script.js [hunk 2]
```diff
@@ -22,7 +24,7 @@ function search_rule(self) { // eslint-disable-line no-unused-vars
         var is_matched_title = $(this).attr("title").toLowerCase().includes(value);
         var is_matched_rule_id = $(this).attr("rule-id").toLowerCase().includes(value);
 
-        var result_of_rule = $(this).attr('result').toLowerCase();
+        var result_of_rule = $(this).attr('result').replace(/ /ug, "-").toLowerCase();
```
Comment (evgenyz) [inline oscap_report/html_report/templates/js/interactive_script.js:27]: I'd prefer to have a function that does such kind of id-lization of a string. With such a string you could always be sure that all ids across the code base would look the same, and if the original was the same so would be corresponding ids.

File: oscap_report/html_report/templates/rule_detail.html (hunk 1 above)
Comment (Honny1) [reply oscap_report/html_report/templates/rule_detail.html]: Span with class `pf-screen-reader` is for accessibility reasons. 
Here is the result: 
image

Comment (evgenyz) [reply oscap_report/html_report/templates/rule_detail.html]: I don't get it. Why do we need a hidden, screen-reader only `Message:` when we already have the same `Message:`, which is readable by screen reader as well as any other client?

Comment (evgenyz) [reply oscap_report/html_report/templates/rule_detail.html]: Summoning @vojtapolasek.

File: oscap_report/html_report/templates/js/interactive_script.js (hunk 2 above)
Comment (evgenyz) [reply oscap_report/html_report/templates/js/interactive_script.js]: I can be a JQuery function, like `$(this).attr('result').as_id_function()`

File: openscap_report/html_report/templates/js/interactive_script.js [hunk 3]
```diff
@@ -8,23 +8,30 @@ var FILTER_TABLE = {
...
+    "result-fix-unsuccessful": true,
     "result-unknown": true,
     "severity-high": true,
     "severity-medium": true,
     "severity-low": true,
     "severity-unknown": true
  };
 
+// eslint-disable-next-line no-extend-native
+String.prototype.as_id_function = function(prefix) {
```
Comment (evgenyz) [inline openscap_report/html_report/templates/js/interactive_script.js:21]: Uh, oh. Not literally. `as_id` should be enough. :)

Comment (evgenyz) [reply openscap_report/html_report/templates/js/interactive_script.js:21]: Actually it should be `.asIdString` acc. to conventions.

File: oscap_report/html_report/templates/rule_detail.html (hunk 1 above)
Comment (Honny1) [reply oscap_report/html_report/templates/rule_detail.html]: Here is a report example [report.zip](https://github.com/OpenSCAP/openscap-report/files/8035470/report.zip). Try rule: `xccdf_org.ssgproject.content_rule_package_aide_installed`

Comment (evgenyz) [reply oscap_report/html_report/templates/rule_detail.html]: We discussed it with @vojtapolasek and agreed to make a global do-over of a11y at some point.

**Pull request merged**